app/rpc/namespaces/eth/api.go

```diff
@@ -1000,17 +1000,18 @@ func (api *PublicEthereumAPI) generateFromArgs(args rpctypes.SendTxArgs) (*evmty
 		gasPrice = big.NewInt(ethermint.DefaultGasPrice)
 	}
 
-	if args.Nonce == nil {
-		// get the nonce from the account retriever and the pending transactions
-		nonce, err = api.accountNonce(api.clientCtx, args.From, true)
-	} else {
-		nonce = (uint64)(*args.Nonce)
-	}
-
+	// get the nonce from the account retriever and the pending transactions
+	nonce, err = api.accountNonce(api.clientCtx, args.From, true)
 	if err != nil {
 		return nil, err
 	}
 
+	if args.Nonce != nil {
+		if nonce != (uint64)(*args.Nonce) {
```

Comment:
only need one if here?
```suggestion
		if args.Nonce != nil && (uint64)(*args.Nonce)  != nonce {
```